### ISSUE TITLE
Type hints from repo/package are not recognized anymore when installed

### ISSUE THREAD

**Issue by Cielquan** [python/mypy#8875]
* **Are you reporting a bug, or opening a feature request?** Bug/Question
* **Please insert below the code you are checking with mypy,
  or a mock-up repro if the source is private. We would appreciate
  if you try to simplify your case to a minimal repro.**
I created a minimized branch on my [testing github repo](https://github.com/Cielquan/python_test/tree/mypy).

* **What is the actual behavior/output?**

When I type check only the `tests/` directory I get an error saying my package was found but that it has neither a stub file nor type hints. But it has a stub file. 

The behavior does not change when I remove the stub file and add type hints or add type hints without removal of the stub file so that I have two sources of type hints.

To see the behavior you just need to `git clone` the above linked repo, be on the `mypy` branch and run `tox`. ` tox` will create a virtualenv, install `mypy` + the package via `poetry` and run three `mypy` commands you can see below incl. output:
```console
mypy run-test: commands[0] | mypy tests/ src/python_test
Success: no issues found in 3 source files
mypy run-test: commands[1] | mypy src/python_test
Success: no issues found in 2 source files
mypy run-test: commands[2] | mypy tests/
tests/test_python_test.py:1: error: Skipping analyzing 'python_test.python_tester': found module but no type hints or library stubs
tests/test_python_test.py:1: note: See https://mypy.readthedocs.io/en/latest/running_mypy.html#missing-imports
tests/test_python_test.py:1: error: Skipping analyzing 'python_test': found module but no type hints or library stubs
Found 2 errors in 1 file (checked 1 source file)
```

When I don't use the `src/` layout the problem obviously does not occur because the test module can directly import the package. This layout I have on the `mypy-no-src` branch.

* **What is the behavior/output you expect?**
I expected that regardless if I call `mypy tests/ src/python_test` or `mypy tests/` both commands pass because the `python_test` package is installed in the same virtualenv `mypy` is called from. 

* **What are the versions of mypy and Python you are using?** mypy=0.770, python 3.8.0
  **Do you see the same issue after installing mypy from Git master?** yes
* **What are the mypy flags you are using? (For example --strict-optional)** None

Is this a bug or am I too dumb to use mypy correctly?

**Comment by ethanhs** [comment]
Ouch this is really unfortunate. 

I actually probably would have missed this if git hadn't failed to check out the `py.typed` file on Windows. The culprit is that there is a stray space at the end of the path to the `py.typed` file, so it wasn't being picked up correctly. I checked and if you remove it everything works as expected.

Thank you for making it so easy to reproduce though. It would have made a bug a lot easier to find :)

**Comment by Cielquan** [comment]
Such a silly mistake on my side. I should stop copy/pasting where it's not necessary. 

Well thanks for your help. I appreciate it. :)
